theseus/core/tests/test_objective.py

```diff
@@ -282,10 +282,10 @@ def _check_error_for_data(v1_data_, v2_data_, error_, error_norm_2_):
 
         # For input_data and also_update
 
-        v1_data = torch.ones(batch_size, dof) * f1 * 0.1
-        v2_data = torch.ones(batch_size, dof) * f2 * 0.1
+        v1_data_new = torch.ones(batch_size, dof) * f1 * 0.1
+        v2_data_new = torch.ones(batch_size, dof) * f2 * 0.1
 
-        input_data = {"v1": v1_data, "v2": v2_data}
+        input_data = {"v1": v1_data_new, "v2": v2_data_new}
         error = objective.error(input_data=input_data, also_update=False)
         error_norm_2 = objective.error_squared_norm(
```

Comment:
Only thing left is to do these test after both `error` and `error_norm_2`. Maybe you can use a common function since the code will mostly be the same for both.

Reply:
@mhmukadam updated the unit test

Reply:
The common function is good. We also need something like this:
```
error = objective.error(...)
_check_variables(...)
...
error_norm_2 = objective.error_squared_norm(...)
_check_variables(...)
```
So we check on both `objective.error` and `objective.error_squared_norm`.

Reply:
Yeah, at this point is probably best to refactor the test code a bit so we run the same checks on `error` and `error_squared_norm` separately. You could write a function that does all checks, and which is parameterized by the function to test (i..e, `error` or `error_squared_norm`), and the expected output of the computation to check against, and then loop it for the two functions.

Reply:
@luisenp  so you're saying about two function `_check_variables()` and `_check_error_for_data()` which will be called inside loop for `error` and `error_squared_norm` right ?

Reply:
You still need to do the checks after both `error` and `error_squared_norm` each. With another commit for this the CI issue should hopefully also go away.

Reply:
@jeffin07 I think you have the right idea. Basically just take the current overall test logic, and repeat it once for `error` and once for `error_squared_norm`.